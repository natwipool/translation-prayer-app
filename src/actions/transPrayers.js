// SET_TRANS_PRAYER
const setTransPrayer = (transPrayers = {}) => ({
  type: 'SET_TRANS_PRAYER',
  transPrayers
});

export default setTransPrayer;