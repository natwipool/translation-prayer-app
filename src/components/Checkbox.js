import React from 'react';

// class Checkbox extends React.Component {
//   constructor(props) {
//     super(props)

//     this.state = {
//       checked: props.lists.find((list) => list === props.precept)
//     }
//   }

//   onAddLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     console.log(this.state.checked)
//     this.props.onAddLists(list)
//   }
  
//   onRemoveLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     this.props.onRemoveLists(list); 
//   }

//   render() {
//     return (
//       <div>
//         <input
//           type="checkbox"
//           name={this.props.precept}
//           value={this.props.precept}
//           checked={this.state.checked}
//           onChange={e => {
//             const list = e.target.value;
//             if (e.target.checked) {
//               this.onAddLists(list)  
//             } else if (!e.target.checked) {
//               this.onRemoveLists(list);  
//             }
//           }}
//         />
//         <label htmlFor={this.props.precept}>
//           {this.props.precept}
//           {
//             this.props.lists.indexOf(this.props.precept) > -1 && (
//               this.props.lists.indexOf(this.props.precept) + 1
//             )
//           }
//         </label>
//       </div>
//     )
//   }
// }

const Checkbox = (props) => (
  <div>
    <input
      type="checkbox"
      name={props.precept}
      value={props.precept}
      checked={props.lists.find((list) => list === props.precept)}
      onChange={e => {
        const list = e.target.value;
        if (e.target.checked) {
          props.onAddLists(list)  
        } else if (!e.target.checked) {
          props.onRemoveLists(list);  
        }
      }}
    />
    <label htmlFor={props.precept}>
      {props.precept}
      {
        props.lists.indexOf(props.precept) > -1 && (
          props.lists.indexOf(props.precept) + 1
        )
      }
    </label>
  </div>
);

export default Checkbox;
