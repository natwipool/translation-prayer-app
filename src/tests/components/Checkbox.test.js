import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../components/Checkbox';
import transPrayers from '../fixtures/transPrayers';

test('should render Checkbox correctly', () => {
  const wrapper = shallow(<Checkbox lists={[]} {...transPrayers[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should get checked value on input checked', () => {
  const onAddListsSpy = jest.fn();
  const wrapper = shallow(<Checkbox lists={[]} {...transPrayers[1]} onAddLists={onAddListsSpy} />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('input[type="checkbox"]').simulate('change', {
    target: { checked: true }
  });
  expect(onAddListsSpy).toHaveBeenCalled();
});

test('should not get checked value on input unchecked', () => {
  const onRemoveListsSpy = jest.fn();
  const wrapper = shallow(<Checkbox lists={[]} {...transPrayers[1]} onRemoveLists={onRemoveListsSpy} />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('input[type="checkbox"]').simulate('change', {
    target: { checked: false }
  });
  expect(onRemoveListsSpy).toHaveBeenCalled();
});