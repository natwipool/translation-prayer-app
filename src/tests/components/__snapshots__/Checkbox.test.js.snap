// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get checked value on input checked 1`] = `
<div>
  <input
    name="บทพิจารณาสังขาร"
    onChange={[Function]}
    type="checkbox"
    value="บทพิจารณาสังขาร"
  />
  <label
    htmlFor="บทพิจารณาสังขาร"
  >
    บทพิจารณาสังขาร
  </label>
</div>
`;

exports[`should not get checked value on input unchecked 1`] = `
<div>
  <input
    name="บทพิจารณาสังขาร"
    onChange={[Function]}
    type="checkbox"
    value="บทพิจารณาสังขาร"
  />
  <label
    htmlFor="บทพิจารณาสังขาร"
  >
    บทพิจารณาสังขาร
  </label>
</div>
`;

exports[`should render Checkbox correctly 1`] = `
<div>
  <input
    name="บทพิจารณาสังขาร"
    onChange={[Function]}
    type="checkbox"
    value="บทพิจารณาสังขาร"
  />
  <label
    htmlFor="บทพิจารณาสังขาร"
  >
    บทพิจารณาสังขาร
  </label>
</div>
`;
